src/app/sidebar/sidebar.controller.js

```diff
@@ -0,0 +1,196 @@
+class CareerPortalSidebarController {
+    constructor($scope, SharedData, $location, SearchService, $timeout, configuration) {
+        'ngInject';
+
+        this.SharedData = SharedData;
+        this.$location = $location;
+        this.$timeout = $timeout;
+        this.SearchService = SearchService;
+        this.configuration = configuration;
+
+        this.locationLimitTo = 8;
+        this.categoryLimitTo = 8;
+
+        this.SearchService.findJobs();
+        this.SearchService.getCountByLocation(this.setLocations());
+        this.SearchService.getCountByCategory(this.setCategories());
+
+        // Set the grid state
+        if (this.configuration && this.configuration.defaultGridState === 'list-view') {
```

Comment:
Consider refactoring this to a switch statement to provide easier maintenance for anyone extending this with other views in the future.

```
switch((this.configuration || {}).defaultGridState) {
    case: 'list-view':
        ...;
    case 'grid-view':
    default:
        ...;
}
```